src/platform/auth.ts

```diff
@@ -86,10 +88,11 @@ export const updatePasswordIfNeeded = (
 	newPassword: string,
 	tx?: Tx,
 ): Promise<boolean> => {
-	return findUser(usernameOrEmail, tx).then((user: AnyObject | undefined) => {
+	return findUser(usernameOrEmail, tx).then(user => {
```

Comment:
Many places that you see typing changes (either adding explicit types or removing `AnyObject`) it revealed that we were possibly checking a null password and as a result I decided to embed the null check in the `comparePassword` method.